src/Shared/Model/ArtifactUri.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model;
+
+using System;
+using System.IO;
+
+public struct ArtifactUri
+{
+    public ArtifactUri(ArtifactType type, Uri uri, string? extension = null)
+    {
+        Type = type;
+        Uri = uri;
+        Extension = extension ?? Path.GetExtension(uri.AbsolutePath);
+    }
+
+    public ArtifactUri(ArtifactType type, string uri, string? extension = null)
+    {
+        Type = type;
+        Uri = new Uri(uri);
+        Extension = extension ?? Path.GetExtension(uri);
+    }
```

Comment:
Should one of these constructors call the other one, to minimize duplication? For example, maybe the constructor that takes a string could turn the string into a `Uri` as part of calling the `Uri` constructor overload?